python/tests/kernel/test_control_operations.py

```diff
@@ -0,0 +1,381 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+import pytest
+import random
+import numpy as np
+
+import cudaq
+from cudaq import spin
+
+
+def test_ctrl_x():
+    """Tests the accuracy of the overloads for the controlled-X gate."""
+
+    @cudaq.kernel
+    def ctrl_x_kernel():
+        qubits = cudaq.qvector(5)
+        controls = cudaq.qvector(2)
+
+        # Overload 1: one control qubit, one target.
+        # 2-qubit controlled operation with control in 0-state.
+        cx(qubits[0], qubits[1])
+        # 2-qubit controlled operation with control in 1-state.
+        x(qubits[2])
+        cx(qubits[2], qubits[3])
+
+        # Overload 2: list of control qubits, one target.
+        # The last qubit in `qubits` is still in 0-state, as is our
+        # `control` register. A controlled gate between them should
+        # have no impact.
+        cx([controls[0], controls[1]], qubits[4])
```

Comment:
I guess this is a question we'll have to raise at the language meeting. I see that you stuck here to the 2 operands idea, but this is still allowing a multi-control X operation, which I was trying to avoid (really this is a `ccx` operation). I think we should only allow 2 qref operands for these new legacy `c` operations.

Reply:
Q: The behavior should be consistent across the modes, builder and MLIR, (backward compatibility breaks?), or only the new mode enforces this check (inconsistency?). 
We can discuss in language meeting.

Reply:
I don't care if it stays for now